packages/application/src/app.ts

```diff
@@ -37,6 +37,11 @@ export class NotebookApp extends JupyterFrontEnd<INotebookShell> {
         this.registerPlugin(plugin);
       }
     }
+
+    this.restored = this.shell.restored
+      .then(() => undefined)
+      .catch(() => undefined);
```

Comment:
Is this needed? Or could `this.shell.restored.then()` be directly used on the line below?

Reply:
Right, it's left from a copy and paste.
But we should keep `this.restored = this.shell.restored`. For example, the table of contents extension waits for `app` to be restored, but needs the Notebook panel to be ready. That's why the `shell` resolves the restored only when the main area has been added.

Reply:
Fixed in https://github.com/jupyter/notebook/pull/6487/commits/52f10ede26493331a2db77fa08d0ce3e65470190